TrackletAlgorithm/TrackHandler.cc

```diff
@@ -1,82 +1,81 @@
+  #pragma HLS array_partition variable=matchesFoundDisk complete dim=0
+  #pragma HLS array_partition variable=masterTrk._barrelStubArray complete dim=0
+  #pragma HLS array_partition variable=trk._barrelStubArray complete dim=0
+  #pragma HLS array_partition variable=masterTrk._diskStubArray complete dim=0
+  #pragma HLS array_partition variable=trk._diskStubArray complete dim=0
+  LOOP_CompareBarrelStubs:
   for (unsigned int barrelStubNum = 0; barrelStubNum < NBarrelStub; barrelStubNum++){
     #pragma HLS unroll
-    auto masterBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
-    auto inputBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
-
-    if((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)){
-      matchesFoundBarrel[barrelStubNum][0] = 1;
-    } else {
-      matchesFoundBarrel[barrelStubNum][0] = 0;
-    }
+    ap_uint<TrackFitType::kTFStubIndexSize> masterBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    ap_uint<TrackFitType::kTFStubIndexSize> inputBarrelStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._barrelStubArray[barrelStubNum][0].range(kBarrelStubIndexSizeMSB, kBarrelStubIndexSizeLSB));
+    
+    ((masterBarrelStubIndex == inputBarrelStubIndex) && (masterBarrelStubIndex > 0)) ? matchesFoundBarrel[barrelStubNum][0] = 1 : matchesFoundBarrel[barrelStubNum][0] = 0;
   }
-  LOOP_CompareTrackDisk:
+  LOOP_CompareDiskStubs:
   for (unsigned int diskStubNum = 0; diskStubNum < NDiskStub; diskStubNum++){
-    #pragma HLS unroll  
-    auto masterDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
-    auto inputDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
-    if ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)){
-      matchesFoundDisk[diskStubNum][0] = 1;
-    } else {
-      matchesFoundDisk[diskStubNum][0] = 0;
-    }
+    #pragma HLS unroll
+    ap_uint<TrackFitType::kTFStubIndexSize> masterDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(masterTrk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+    ap_uint<TrackFitType::kTFStubIndexSize> inputDiskStubIndex = ap_uint<TrackFitType::kTFStubIndexSize>(trk._diskStubArray[diskStubNum][0].range(kDiskStubIndexSizeMSB, kDiskStubIndexSizeLSB));
+
+    ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)) ? matchesFoundDisk[diskStubNum][0] = 1 : matchesFoundDisk[diskStubNum][0] = 0;
```

Comment:
Likewise, this can be more compact:
```suggestion
    matchesFoundDisk[diskStubNum][0] = ((masterDiskStubIndex == inputDiskStubIndex) && (masterDiskStubIndex > 0)) ? 1 : 0;
```